src/states/AnimatingState.ts

```diff
@@ -1,20 +1,38 @@
 import State from "./State";
 import { STATE_TYPE, EVENTS } from "../consts";
 import { FlickingContext } from "../types";
+import { circulate } from "../utils";
 
 class AnimatingState extends State {
   public readonly type = STATE_TYPE.ANIMATING;
   public readonly holding = false;
   public readonly playing = true;
 
-  public onEnter(e: any): void {
-    super.onEnter(e);
+  public onHold(e: any, { viewport, triggerEvent, transitTo }: FlickingContext): void {
+    const scrollArea = viewport.getScrollArea();
+    const scrollAreaSize = viewport.getScrollAreaSize();
+    const loopCount = Math.floor((this.lastPosition + this.delta - scrollArea.prev) / scrollAreaSize);
+
+    let targetPanel = this.targetPanel;
+    if (loopCount !== 0 && targetPanel) {
+      const cloneCount = viewport.panelManager.getCloneCount();
+      const originalTargetPosition = targetPanel.getPosition();
+
+      // cloneIndex is from -1 to cloneCount - 1
+      const newCloneIndex = circulate(targetPanel.getCloneIndex() - loopCount, -1, cloneCount - 1);
```

Comment:
@WoodNeck 
Is there a chance that newCloneIndex will get -1?

Reply:
As written in comment, cloneIndex is from -1 to cloneCount - 1
It's because original panel's cloneIndex is -1, and clone's cloneIndex starts from 0.
I've wrote code in that way as it's more convenient to manage clones like that.

So, newCloneIndex can be -1, if virtual panel is pointing original panel.